scene/resources/environment.cpp

```diff
@@ -891,23 +892,41 @@ float Environment::get_adjustment_saturation() const {
 	return adjustment_saturation;
 }
 
-void Environment::set_adjustment_color_correction(const Ref<Texture2D> &p_ramp) {
-	adjustment_color_correction = p_ramp;
+void Environment::set_adjustment_color_correction(Ref<Texture> p_color_correction) {
+	adjustment_color_correction = p_color_correction;
+	Ref<GradientTexture> grad_tex = p_color_correction;
+	if (grad_tex.is_valid()) {
+		if (!grad_tex->is_connected(CoreStringNames::get_singleton()->changed, callable_mp(this, &Environment::_update_adjustment))) {
+			grad_tex->connect(CoreStringNames::get_singleton()->changed, callable_mp(this, &Environment::_update_adjustment));
+			if (grad_tex->get_gradient().is_valid() && !grad_tex->get_gradient()->is_connected(CoreStringNames::get_singleton()->changed, callable_mp(this, &Environment::_update_adjustment))) {
+				grad_tex->get_gradient()->connect(CoreStringNames::get_singleton()->changed, callable_mp(this, &Environment::_update_adjustment));
+			}
+		}
+	}
```

Comment:
```suggestion
	if (grad_tex.is_valid()) {
		if (!grad_tex->is_connected(CoreStringNames::get_singleton()->changed, callable_mp(this, &Environment::_update_adjustment))) {
			grad_tex->connect(CoreStringNames::get_singleton()->changed, callable_mp(this, &Environment::_update_adjustment));
		}
	}
```

You are right. The RID needs to be updated when the GradientTexture changes. But it doesn't need to be updated everytime the Gradient changes.